### ISSUE TITLE
SC2169 is wrong about (d)ash echo flag -e

### ISSUE THREAD

**Issue by yuvadm** [koalaman/shellcheck#1006]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2169
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/ash
echo -e "hello\nworld"
```

#### Here's what shellcheck currently says:
```
^-- SC2169: In dash, echo flags besides -n not supported.
```

#### Here's what I wanted or expected to see:
The `-e` flag works just fine, in `ash` at least (which should be identical to `dash`):

```sh
$ ash
BusyBox v1.25.1 () built-in shell (ash)
$ echo -e "hello\nworld"
hello
world
```
CC: #853

**Comment by koalaman** [comment]
`dash` does not support `echo -e`:

```
$ dash -c 'echo -e "hello\nworld"'
-e hello
world
```

ShellCheck doesn't support `ash` at this time, so you can either ignore the warning or rewrite it in a `sh` or `dash` compatible way (with `printf`).
